go-gemini/cmd/web/main.go

```diff
@@ -48,28 +48,28 @@ func generateHandler(w http.ResponseWriter, r *http.Request, model *genai.Genera
 		return
 	}
 
-	// Generate the response and aggregate the streamed response.
-	iter := model.GenerateContentStream(r.Context(), genai.Text(prompt), genai.ImageData("jpeg", contents))
-	for {
-		resp, err := iter.Next()
-		if err == iterator.Done {
-			break
-		}
-		if err != nil {
-			log.Printf("Error generating content: %v\n", err)
-			http.Error(w, "Error: unable to generate content", http.StatusInternalServerError)
-			return
-		}
-		if resp == nil {
-			continue
-		}
-		for _, cand := range resp.Candidates {
-			if cand.Content != nil {
-				for _, part := range cand.Content.Parts {
-					fmt.Fprint(w, part)
-				}
-			}
-		}
+	content := []llms.MessageContent{
+		{
+			Role: schema.ChatMessageTypeHuman,
+			Parts: []llms.ContentPart{
+				llms.BinaryPart("image/jpeg", contents),
+				llms.TextPart(prompt),
+			},
+		},
+	}
+
+	_, err = llm.GenerateContent(r.Context(), content,
+		llms.WithModel("gemini-pro-vision"),
+		llms.WithMaxTokens(500), // default of 256 is not enough.
```

Comment:
This is an odd comment to have as part of a template...

Do you think the default max token limit is too low? Maybe I can just raise it

Reply:
For this recipe generator, the output was always cut off when using the default. I think if we have any concern of it being close to the limit, having this option here is at least nice to make it easy to notice when things go wrong.

It was also interesting because if I didnt use the streaming func, then the response would have empty contents, whereas with streaming I would be able to get a partial response (which is slightly more subtly broken).

Reply:
Fair enough.
It should specify a finish reason when reaching max token limit, by the way. 
https://pkg.go.dev/github.com/google/generative-ai-go@v0.10.0/genai#FinishReasonMaxTokens

Probably out of scope for this sample - up to you

Reply:
I guess maybe a bug, but I also noticed the finish reason was only FinishReasonMaxTokens if I did not use StreamingFunc ( I think it was unspecified when using StreamingFunc but I will double check).

I think it probably makes sense to log the finish reason if its not FinishReasonStop, so can add that in so we actually use the response as well.

Reply:
Ended up skipping checking the stop reason because there is a bug. Will file an issue.